pyfolio/perf_attrib.py

```diff
@@ -97,6 +98,37 @@ def perf_attrib(returns, positions, factor_returns, factor_loadings,
             2017-01-01  0.249087  0.935925        1.185012          1.185012
             2017-01-02 -0.003194 -0.400786       -0.403980         -0.403980
     """
+
+    missing_stocks = positions.columns.difference(
+        factor_loadings.index.levels[1]
+    )
+
+    # cash will not be in factor_loadings
+    missing_stocks = missing_stocks.drop('cash')
+
+    if len(missing_stocks) > 0:
+
+        warnings.warn("Could not find factor loadings for the following "
+                      "stocks: {}. Ignoring for exposure calculation and "
+                      "performance attribution.".format(list(missing_stocks)))
+
+        positions = positions.drop(missing_stocks, axis='columns')
+
+    missing_factor_loadings_index =\
+        positions.index.difference(factor_loadings.index.levels[0])
```

Comment:
Rather than the difference, we want to subset of dts in positions, but not in factor_loadings (i.e. assymetrical).

Reply:
yup, difference is symmetrical
```
> pd.Index(factor_loadings.index.get_level_values(1).unique()).difference(positions.columns)
Index([], dtype='object')
> positions.columns.difference(factor_loadings.index.get_level_values(1).unique())
Index([u'TLT', u'cash'], dtype='object')
```